crates/ruff_db/src/vendored.rs

```diff
@@ -256,6 +283,62 @@ mod tests {
         VendoredFileSystem::new(&MOCK_ZIP_ARCHIVE).unwrap()
     }
 
+    #[test]
+    fn manual_debug_implementation() {
+        assert_snapshot!(
+            format!("{:?}", mock_typeshed()),
+            @"VendoredFileSystem { inner: VendoredFileSystemInner(Mutex { data: RefCell { value: VendoredZipArchive(<4 paths>) }, poisoned: false, .. }) }"
+        );
```

Comment:
I'm slightly confused. Why does the manual debug still render the inner `Mutex` considering that it is implemented on `VendoredFileSystem`

Reply:
Agreed, though for debugging purposes I think it would be useful to know if the inner mutex were unexpectedly poisoned for some reason. I've just reworked it a bit to expose more information about the inner zip files (when the alternate debug representation has been specified) but expose less about the intermediate types between the zip archive and the `VendoredFileSystem` struct... let me know what you think!